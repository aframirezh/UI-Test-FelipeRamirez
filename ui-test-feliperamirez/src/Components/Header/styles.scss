$headerColor: #fff;
$headerHoverColor: gray;
$HeaderGradientFrom: rgba(0,0,0,0.7);
$HeaderGradientTo: rgba(0,0,0,0);
$headerSearchColor: #cfcdcd;

header { 
    color: $headerColor;
    background: linear-gradient(to bottom , $HeaderGradientFrom, $HeaderGradientTo);
	display:flex;
    height: 100px; 
    overflow: hidden;
	width: 100%;
	
	div:first-child {
		flex-grow:8;
		padding-left: 60px;
	}

	div h2 {
		padding-top: 0px;
		font-weight:400;
		filter: blur(0.5px);
	}

	div img {
		padding-top: 20px;
	}
}

a:hover {
	color: $headerHoverColor;
}

ul {
    color:$headerColor;
	list-style-type:none;
	display:flex;
	flex-flow: row wrap;
	margin:0;
	padding-top: 20px;
	width:400px;
	
	li {
		flex-grow:1;
		text-align:center;
		width:100px;

		a {
			font-size:.9em;
			text-decoration: none;
			color: $headerColor;
		}
	}
}

#magnify {
	cursor:pointer;
}

#search {
	background: $headerSearchColor url(./images/magnify.png) no-repeat 5px 5px;
	border-radius:10px;
	border:none;
	outline:none;
	color:$headerColor;
	padding: 0px;
	display:none;
	height: 30px;
}

@media only screen and (max-width:800px) {

	header {
        justify-content: space-around;
        width: 100%;
		height: auto;
		
		div h2 {
			padding-left: 10px;
			font-weight:500;
			text-align: left;
		}

		div:first-child {
			flex-grow:8;
			padding-left: 10px;
		}

		div img {
			padding-top: 20px;
		}
	}

	.container {
		padding:50px;
	}

	ul {
		list-style-type:none;
		display:flex;
		flex-flow: row wrap;
		margin:0;
		padding-top: 20px;

		li {
			flex-grow:1;
			text-align:center;
			width:100px;
			font-size:.9em;
		}
	}
}

@media only screen and (max-width:583px) {

	header {
		flex-direction: row;
        padding-bottom: 15px;
        width: 100%;
		height: auto;
		
		div h2 {
			margin: 0 auto;
			font-weight:500;
			display:block;
			padding-top: 10px;
			text-align: center;
		}

		div:nth-child(2) {
			align-self:flex-end;
			width:22px;
		}
	}

	.child {
		flex-grow:1;
		text-align: center;
	}

	ul {
        width:100%;
        height: 20px;
		margin-top: 0px;

		li {
			width:20%;
			padding: 0px;
		}
	}

	#magnify {
        display:none; 
        font-size-adjust: 58;
    }

	#search {
		display:block;
		width:80%;
		margin: 6px auto 0 auto;
	}    
}