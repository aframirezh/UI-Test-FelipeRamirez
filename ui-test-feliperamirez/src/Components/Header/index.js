import React from 'react';
import magnify from './images/magnify.png';
import './styles.scss';

function Header () {
  return (
    <header>
      <div className="child">
        <h2>Rule of Thumb.</h2>
      </div>
      <ul className="child">
        <li><a href="#">Past Trials</a></li>
        <li><a href="#">How It Works</a></li>
        <li><a href="#">Log In/Sign Up</a></li>
      </ul>

      <div className="child">
        <img src={magnify} id="magnify" ></img>
        <input type="text" name="search" id="search"></input>
      </div>
    </header>
  );
}

export default Header;