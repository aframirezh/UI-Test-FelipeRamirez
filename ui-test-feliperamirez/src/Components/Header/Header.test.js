import React from 'react';
import { render } from '@testing-library/react'
import Header from '.';

describe("Header rendering", () => {
    const comp = render(<Header/>)
    it('Header Correct rendering', () => {
        expect(comp.asFragment()).toMatchSnapshot();
    })
});