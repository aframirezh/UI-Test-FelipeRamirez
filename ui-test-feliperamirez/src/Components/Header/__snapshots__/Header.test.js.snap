// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header rendering Header Correct rendering 1`] = `
<DocumentFragment>
  <header>
    <div
      class="child"
    >
      <h2>
        Rule of Thumb.
      </h2>
    </div>
    <ul
      class="child"
    >
      <li>
        <a
          href="#"
        >
          Past Trials
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          How It Works
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          Log In/Sign Up
        </a>
      </li>
    </ul>
    <div
      class="child"
    >
      <img
        id="magnify"
        src="magnify.png"
      />
      <input
        id="search"
        name="search"
        type="text"
      />
    </div>
  </header>
</DocumentFragment>
`;
